### PR TITLE
Session refinement

### PR DESCRIPTION
**Pull request by jdillon**
Related:
* https://github.com/sonatype/nexus-pro/pull/503

Enables permissive mode for basic-auth so that anonymous can access content if it has the right permissions.
Adds more shiro components to control how session bits are setup.
Disables saving anonymous subjects, pending further optimization to avoid calling deep into shiro due to use of buildSubject() helper, but will sort that out later.
Other logging and clean up

### CODE REVIEW

**Comment by adreghiciu** [comment]
+1

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF97-1 :white_check_mark:

File: components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java [hunk 1]
```diff
@@ -56,7 +64,7 @@
  *
  * @since 2.6.1
  *
- * @see SecurityModule
+ * @see TestSecurityModule
```
Comment (mcculls) [inline components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java]: I don't think this reference makes sense now and should be removed

**Comment by mcculls** [comment]
+1

File: components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java (hunk 1 above)
Comment (jdillon) [inline components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java]: I left it here until TestSecurityModule was removed, which will be soon... though its still used no?  I can remove the @see ref now though.

Comment (mcculls) [inline components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java]: I'll be removing it in my PR, so feel free to go ahead and merge this as-is and I'll clean it up later.

Comment (jdillon) [inline components/nexus-security/src/main/java/org/sonatype/nexus/security/WebSecurityModule.java]: @mcculls I'm included to go with that for now ;-)  BTW glad you are removing it!

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**